src/components/books/booksSlice.js

```diff
@@ -0,0 +1,21 @@
+// booksSlice.js
+
+import { createSlice } from '@reduxjs/toolkit';
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState: [],
+  reducers: {
+    bookAdded(state, action) {
+      state.push(action.payload);
+    },
+    bookRemoved(state, action) {
+      const index = state.findIndex((book) => book.id === action.payload);
```

Comment:
- [ ] [OPTIONAL] in case action.type is REMOVE_BOOK the function uses the ES6 filter() method to create a new array, which will not contain the book you want to remove from the store (filter by the id key - i.e.: return state.filter(book => book.id !== id),  kindly consider the immutability of the state, so the state is not overwritten instead it should contain the previous state and the new one